pkg/storage/iterators.go

```diff
@@ -61,7 +60,7 @@ func (c *combinedIterator[T]) Next(ctx context.Context) (T, error) {
 	if err != nil {
 		if errors.Is(err, ErrIteratorDone) {
 			c.pending = c.pending[1:]
-			c.done = append(c.done, iter)
+			iter.Stop() // clean up before dropping the reference
```

Comment:
Could this be moved to a different PR? :)

Reply:
Sigh, yes.